### PR TITLE
11818 improve bag api

### LINKED ISSUE #11818: Improve Bag api: adding addAll:
```
Bag new addAll: { 'ab' . 'b' . 'ab' . 'c'}; keys 
>>> #('ab' 'b' 'c')
```

is missing.

### PR DESCRIPTION
**Pull request by Ducasse**
fixes: #11818 improve bag api

### CODE REVIEW

**Comment by Ducasse** [comment]
I relaunched the build because this is super strange that it broke. The code works in my image and it is super simple.

**Comment by Ducasse** [comment]
I do not understand the problem

**Comment by MarcusDenker** [comment]
very strange build problem:

```
#errorMessage was sent to nil
UndefinedObject(Object)>>doesNotUnderstand: #errorMessage
UndefinedObject>>doesNotUnderstand: #errorMessage
RuntimeSyntaxError>>errorMessage
RuntimeSyntaxError>>messageText
HDTestReport>>writeException:asNode:using:
HDTestReport>>writeException:of:asNode:
HDTestReport>>writeError:of:
HDTestReport>>recordError:duringTest:
RuntimeSyntaxError(Error)>>recordResultOf:inHDTestReport:
```

**Comment by Ducasse** [comment]
Yes I do not really understand why such a simple change is generating such problems.

**Review by guillep — APPROVED** [review]
ok

**Comment by guillep** [comment]
Let's wait until the CI runs

File: src/Collections-Unordered/Bag.class.st [hunk 1]
```diff
@@ -171,7 +171,16 @@ Bag >> includes: anObject [
 ]
 
 { #category : #enumerating }
-Bag >> keysAndValuesDo: aTwoArgBlock [
+Bag >> keys [
+	"Returns the keys of the receiver."
+	
+	"Bag new add: #a; add: #a; add: #b; keys >>> #(#a #b)"
```
Comment (guillep) [inline src/Collections-Unordered/Bag.class.st]: This doc comment is missing a parenthesis and seems to be breaking the test run in the CI

**Comment by guillep** [comment]
This is strange, this failed consistently in three builds with a Reflectivity crash:

```
0x16aecdea8 M ReflectiveMethod>run:with:in: 0x10c67dc50: a(n) ReflectiveMethod
0x16aecdf00 M ReflectivityControlTest>testAfterSequence 0x10b6fd498: a(n) ReflectivityControlTest
```

**Pull request closed without merging**